mpas_analysis/ocean/meridional_overturning_circulation.py

```diff
@@ -463,62 +526,10 @@ def write_file(dsMOCTimeSeries):
     continueOutput = os.path.exists(outputFileTseries)
     if continueOutput:
         print '   Read in previously computed MOC time series'
-        # read in what we have so far
-        dsMOCTimeSeries = xr.open_dataset(outputFileTseries,
-                                          decode_times=False)
-        # force loading and then close so we can overwrite the file later
-        dsMOCTimeSeries.load()
-        dsMOCTimeSeries.close()
-    else:
-        # create a new data set from a dictionary
-        dictonary = {
-            'Time': {'dims': ('Time'),
-                     'data': []},
-            'mocAtlantic26': {'dims': ('Time'),
-                              'data': []}}
-        dsMOCTimeSeries = xr.Dataset.from_dict(dictonary)
-
-    newEntries = False
-    first = True
-    for tIndex in range(ds.Time.size):
-        if ds.Time[tIndex].values in dsMOCTimeSeries.Time.values:
-            # we have it already, so no need to recompute
-            continue
-        if first:
-            print '   Process and save MOC time series'
-            first = False
-        date = days_to_datetime(ds.Time[tIndex], calendar=calendar)
-        print '     date: {:04d}-{:02d}'.format(date.year, date.month)
-        horizontalVel = ds.avgNormalVelocity[tIndex, :, :].values
-        verticalVel = ds.avgVertVelocityTop[tIndex, :, :].values
-        velArea = verticalVel * areaCell[:, np.newaxis]
-        transportZ = _compute_transport(maxEdgesInTransect,
-                                        transectEdgeGlobalIDs,
-                                        transectEdgeMaskSigns,
-                                        nVertLevels, dvEdge,
-                                        refLayerThickness,
-                                        horizontalVel)
-        mocTop = _compute_moc(latAtlantic, nVertLevels, latCell,
-                              regionCellMask, transportZ, velArea)
-        mocAtlantic26 = np.amax(mocTop[:, indlat26])
-
-        dictonary = {'Time': {'dims': ('Time'), 'data': [ds.Time[tIndex]]},
-                     'mocAtlantic26': {'dims': ('Time'),
-                                       'data': [mocAtlantic26]}}
-        # add the new time entry to the data set
-        dsMOCTimeSeries = xr.concat([dsMOCTimeSeries,
-                                     xr.Dataset.from_dict(dictonary)],
-                                    dim='Time')
-        newEntries = True
-        if date.month == 12:
-            # each year, save the data set for safe keeping
-            print '   saving progress.'
-            dsMOCTimeSeries = write_file(dsMOCTimeSeries)
-            newEntries = False
-
-    if newEntries:
-        print '   saving final result.'
-        dsMOCTimeSeries = write_file(dsMOCTimeSeries)
+
+    dsMOCTimeSeries = time_series.cache_time_series(
+        ds.Time.values,  compute_moc_time_series_part, outputFileTseries,
+        calendar, yearsPerCacheUpdate=1,  printProgress=False)
```

Comment:
`yearsPerCacheFile` applies to climatologies.  There could, in theory, be a similar option for the frequency of updating time series in the config file.  Note that time series always get cached in a single file, so it is much less important to the user whether this cache file gets update frequently or rarely.  The only case in which this matters is if the job gets killed part way through computing the time series.

In the MOC, the time series is expensive to compute so we want to update often.  For all other time series, it doesn't make sense to update as often.  So I just hard coded these numbers.  I don't foresee users needing to know about these details or needing to change these frequencies.  The same is not true of climatologies, where the intermediate data might be useful and the frequency of caching might strongly affect performance.